rskj-core/src/test/java/co/rsk/peg/bitcoin/P2shP2wshErpCustomRedeemScriptBuilderTest.java

```diff
@@ -0,0 +1,208 @@
+            // csv is above the maximum allowed
+            Arguments.of(RedeemScriptCreationException.class, defaultKeys, DEFAULT_THRESHOLD, emergencyKeys, ERP_THRESHOLD, surpassingMaxCsvValue)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("providePubKeysAndThresholds")
+    void of_shouldHaveTheCorrectRedeemScript(List<BtcECKey> defaultKeys, int threshold) {
+        // Act
+        Script redeemScript = builder.of(
+            defaultKeys, threshold, emergencyKeys, ERP_THRESHOLD, CSV_VALUE
+        );
+
+        // Assert
+        byte[] p2shp2wshErpCustomRedeemScriptProgram = redeemScript.getProgram();
+
+        // redeemScript - First opcode should be the OP_NOTIF
+        int opNotIfIndex = 0;
+        byte actualOpCode = p2shp2wshErpCustomRedeemScriptProgram[opNotIfIndex];
+
+        assertEquals((byte) ScriptOpCodes.OP_NOTIF, actualOpCode);
+
+        // defaultCustomRedeemScript - Second byte should be the PubKey
+        int pubKeyIndex = opNotIfIndex + 1; //Second byte should have the pubKey size
+        byte actualPubKeyLength = p2shp2wshErpCustomRedeemScriptProgram[pubKeyIndex++];
+        List<BtcECKey> reversedDefaultKeys = Lists.reverse(defaultKeys);
+        byte[] expectedFederatorPubKey = reversedDefaultKeys.get(0).getPubKey();
+
+        assertEquals(expectedFederatorPubKey.length, actualPubKeyLength);
+
+        for (byte expectedCharacterPubKey : expectedFederatorPubKey) {
+            assertEquals(expectedCharacterPubKey, p2shp2wshErpCustomRedeemScriptProgram[pubKeyIndex++]);
+        }
+
+        // defaultCustomRedeemScript - Third opcode should be the OP_CHECKSIG for the PubKey1
+        int opCheckSigIndex = pubKeyIndex;
+        actualOpCode = p2shp2wshErpCustomRedeemScriptProgram[opCheckSigIndex];
+
+        assertEquals((byte) ScriptOpCodes.OP_CHECKSIG, actualOpCode);
+
+        reversedDefaultKeys = reversedDefaultKeys.subList(1, reversedDefaultKeys.size());
+        for (BtcECKey pubKey : reversedDefaultKeys) {
+            // defaultCustomRedeemScript - Forth opcode should be the OP_SWAP for the PubKey1
+            int opSwapIndex = opCheckSigIndex + 1;
+            actualOpCode = p2shp2wshErpCustomRedeemScriptProgram[opSwapIndex];
+
+            assertEquals((byte) ScriptOpCodes.OP_SWAP, actualOpCode);
+
+            pubKeyIndex = opSwapIndex + 1;
+            actualPubKeyLength = p2shp2wshErpCustomRedeemScriptProgram[pubKeyIndex++];
+            expectedFederatorPubKey = pubKey.getPubKey();
+
+            assertEquals(expectedFederatorPubKey.length, actualPubKeyLength);
+
+            for (byte expectedCharacterPubKey : expectedFederatorPubKey) {
+                assertEquals(expectedCharacterPubKey, p2shp2wshErpCustomRedeemScriptProgram[pubKeyIndex++]);
+            }
+
+            // defaultCustomRedeemScript - Third opcode should be the OP_CHECKSIG for the PubKey1
```

Comment:
```suggestion
            // after pubkey we should have OP_CHECKSIG opcode
```
something like that? wdyt? (btw i think you copy-pasted this comment but is not related to pub key **1** anymore)

Reply:
fixed it, check it